packages/react-data-grid/src/HeaderRow.tsx

```diff
@@ -103,11 +107,13 @@ export default class HeaderRow extends React.Component<Props> {
 
     for (let i = 0, len = getSize(columns); i < len; i++) {
       const column = getColumn(columns, i);
+      const key = this.getHeaderCellKey(column);
       const renderer = column.key === 'select-row' && rowType === HeaderRowType.FILTER ? <div /> : this.getHeaderRenderer(column);
 
       const cell = (
         <HeaderCell
-          ref={node => this.cells[i] = node}
+          key={key}
+          ref={node => node ? this.cells.set(key, node) : this.cells.delete(key)}
```

Comment:
Not sure if this is a problem, but don't we need to check if the key value pair already exists in the map before setting it? The result behavior is same, just wondering if React directly pass the node (same obj each time) or create new reference and pass down.

Reply:
No? It doesn't matter either way.